src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -0,0 +1,178 @@
+        /// <param name="traceFileName">trace file name</param>
+        /// <param name="guiApp">true: app under test is a GUI app, false: console</param>
+        /// <param name="skipProfileIteration">true: skip full results iteration</param>
+        /// <param name="reportJsonPath">path to save report json</param>
+        /// <returns></returns>
+        static int Main(string appExe,
+                        MetricType metricType,
+                        string scenarioName,
+                        string traceFileName,
+                        bool processWillExit = false,
+                        int iterations = 5,
+                        int timeout = 60,
+                        int measurementDelay = 15,
+                        string appArgs = "",
+                        string logFileName = "",
+                        string workingDir = "",
+                        bool warmup = true,
+                        bool guiApp = true,
+                        bool skipProfileIteration = false,
+                        string reportJsonPath = "")
+        {
+            Logger logger = new Logger(String.IsNullOrEmpty(logFileName) ? $"{appExe}.startup.log" : logFileName);
+            static void checkArg(string arg, string name)
+            {
+                if (String.IsNullOrEmpty(arg))
+                    throw new ArgumentException(name);
+            };
+            checkArg(scenarioName, nameof(scenarioName));
+            checkArg(appExe, nameof(appExe));
+            checkArg(traceFileName, nameof(traceFileName));
+
+            bool failed = false;
+            logger.Log($"Running {appExe} (args: \"{appArgs}\")");
+            var procHelper = new ProcessHelper()
+            {
+                ProcessWillExit = processWillExit,
+                Timeout = timeout,
+                MeasurementDelay = measurementDelay,
+                Executable = appExe,
+                Arguments = appArgs,
+                WorkingDirectory = workingDir,
+                GuiApp = guiApp
+            };
+            Util.Init();
+
+            if (warmup)
+            {
+                procHelper.Run();
+            }
+
+            string kernelTraceFile = Path.ChangeExtension(traceFileName, "perflabkernel.etl");
+            string userTraceFile = Path.ChangeExtension(traceFileName, "perflabuser.etl");
+            IParser parser = null;
+            switch (metricType)
+            {
+                case MetricType.TimeToMain:
+                    parser = new TimeToMainParser();
+                    break;
+                case MetricType.GenericStartup:
+                    parser = new GenericStartupParser();
+                    break;
+                    //case MetricType.WPF:
+                    //    parser = new WPFParser();
+                    //    break;
+            }
+            using (var kernel = new TraceEventSession(KernelTraceEventParser.KernelSessionName, kernelTraceFile))
```

Comment:
You might be able to create a single trace event session for kernel and user data if you target recent Windows OS (there are some comments on TraceEvent about it), then you only need to deal with a single file and skip the merge of etl.

Reply:
Will take a look. Would clean things up a bit. (Though I note PerfView doesn't do this either..)

Reply:
I'd be interested to know what version of the OS allows for this.  Unless it goes pretty far back (like Win7) I wouldn't make this change, as inevitably, we'll want to run this on old operating systems some day.